src/map.cpp

```diff
@@ -4565,91 +4565,128 @@ void map::debug()
-   // While viewing indoor areas use lightmap model
-   if (!is_outside(realx, realy)) {
-    sight_range = natural_sight_range;
-   // Don't display area as shadowy if it's outside and illuminated by natural light
-   //and illuminated by source of light
-   } else if (this->light_at(realx, realy) > LL_LOW || dist <= light_sight_range) {
-    low_sight_range = std::max(g_light_level, natural_sight_range);
-   }
+    g->reset_light_level();
 
-   // I've moved this part above loops without even thinking that
-   // this must stay here...
-   int real_max_sight_range = light_sight_range > max_sight_range ? light_sight_range : max_sight_range;
-   int distance_to_look = DAYLIGHT_LEVEL;
-
-   bool can_see = pl_sees( realx, realy, distance_to_look );
-   lit_level lit = light_at(realx, realy);
-
-   // now we're gonna adjust real_max_sight, to cover some nearby "highlights",
-   // but at the same time changing light-level depending on distance,
-   // to create actual "gradual" stuff
-   // Also we'll try to ALWAYS show LL_BRIGHT stuff independent of where it is...
-   if (lit != LL_BRIGHT) {
-       if (dist > real_max_sight_range) {
-           int intLit = (int)lit - (dist - real_max_sight_range)/2;
-           if (intLit < 0) intLit = LL_DARK;
-           lit = (lit_level)intLit;
-       }
-   }
-   // additional case for real_max_sight_range
-   // if both light_sight_range and max_sight_range were small
-   // it means we really have limited visibility (e.g. inside a pit)
-   // and we shouldn't touch that
-   if (lit > LL_DARK && real_max_sight_range > 1) {
-       real_max_sight_range = distance_to_look;
-   }
+    update_visibility_variables();
 
-   if ((bio_night_active && dist < 15 && dist > natural_sight_range) || // if bio_night active, blackout 15 tile radius around player
-       dist > real_max_sight_range ||
-       (dist > light_sight_range &&
-         (lit == LL_DARK ||
-         (u_sight_impaired && lit != LL_BRIGHT) ||
-          !can_see))) {
-    if (u_is_boomered)
-     mvwputch(w, realy+getmaxy(w)/2 - center.y, realx+getmaxx(w)/2 - center.x, c_magenta, '#');
-    else
-         mvwputch(w, realy+getmaxy(w)/2 - center.y, realx+getmaxx(w)/2 - center.x, c_dkgray, '#');
-   } else if (dist > light_sight_range && u_sight_impaired && lit == LL_BRIGHT) {
-    if (u_is_boomered)
-     mvwputch(w, realy+getmaxy(w)/2 - center.y, realx+getmaxx(w)/2 - center.x, c_pink, '#');
-    else
-     mvwputch(w, realy+getmaxy(w)/2 - center.y, realx+getmaxx(w)/2 - center.x, c_ltgray, '#');
-   } else if (dist <= u_clairvoyance || can_see) {
-    drawsq(w, g->u, realx, realy, false, true, center.x, center.y,
-           (dist > low_sight_range && LL_LIT > lit) ||
-           (dist > sight_range && LL_LOW == lit),
-           LL_BRIGHT == lit);
-   } else {
-    mvwputch(w, realy+getmaxy(w)/2 - center.y, realx+getmaxx(w)/2 - center.x, c_black,' ');
-   }
-  }
- }
-    g->draw_critter( g->u, center );
+    lit_level ll;
```

Comment:
I know, the old code had it that way, but this variable should not be defined outside of the loops, because it is not used outside the inner loop. Declaring it in the loop makes clear that it is only valid there.